### DISCUSSION TITLE
Save file to server after download

### DISCUSSION THREAD

**Discussion by craigmarcussen** [Hopding/pdf-lib#1564] [Q&A]
Hi all,

First off, let me say that this is an incredible project, and I am very impressed with the implementation and the easy of use. Also, the documentation is excellent. and the Fiddles provided are quite helpful.

I have pdf-lib working perfectly on my site to transfer online form data to a PDF template, and custom name/save that PDF file to the users' desktop. My question: is there a way to save that file to a folder on the same server during the process? Currently, I have to then have the user upload the file manually to the server after the PDF is downloaded. I would rather have this happen automatically, all in one step.

Thank you in advance for any help/suggestions, and for a truly terrific plugin.
Craig

**Comment by laszlo1337** [comment]
Just spitballing here, but maybe:
```
	const pdfBytes = await pdfDoc.save()
	await writeFile(pdfFilePath, pdfBytes) //nodejs fs
	download(pdfBytes, "out.pdf", "application/pdf");
```
